libsolidity/parsing/Parser.cpp

```diff
@@ -1651,7 +1653,7 @@ ASTPointer<Expression> Parser::parseLeftHandSideExpression(
 		nodeFactory.markEndPosition();
 		auto expressionType = nodeFactory.createNode<ElementaryTypeName>(
 			ElementaryTypeNameToken(Token::Address, 160, 0),
-			boost::make_optional(StateMutability::Payable)
+			std::optional{StateMutability::Payable}
```

Comment:
It looks like the Emscripten build fails here.

Reply:
The Emscripten's c++ library does not seem to provide the deduction guide for std::optional, that's however [part of C++17](https://en.cppreference.com/w/cpp/utility/optional/deduction_guides).